### ISSUE TITLE
Simple testcase for DEF parser

### ISSUE THREAD

**Issue by nicolati** [jefftrull/EDASkel#3]
Hi Jeff,
I'm submitting a simple testcase for the DEF parser, extracted from a real testcase.
Actually the DEF parser is not able to parse it.
Another thing, the supported constructs depend obviously on the version specified in the DEF. This one is 5.6 .
[test.def.zip](https://github.com/jefftrull/EDASkel/files/122136/test.def.zip)

**Comment by jefftrull** [comment]
This is very helpful, thank you.  Real world DEF files are hard to find if you are not working directly in IC design.

**Comment by nicolati** [comment]
Yeah, I know. You should support also for DEF the versioning, since commands change depending on it.

Fra

**Comment by jefftrull** [comment]
I have created a `nicolati_bugs` branch and am slowly implementing the necessary features to support the missing syntax there.

**Comment by nicolati** [comment]
Very good! :)
I will keep checking! :)

Fra

**Comment by jefftrull** [comment]
I'm finally working on this again :)

I noticed that U265 and U267 are used in the NETS section but are not defined as components, which is supposed to be an error.  I will make something up for now but if you want to supply an improved test case please feel free.

**Comment by nicolati** [comment]
Hi Jeff,
I don't remember anymore which kind of DEF I have provided to you! :)
If the COMPONENTS section is totally missing, this is a valid DEF, but, if you don't instantiate the 2 missing components somehow (Verilog read, for instance), it has to be consider of course as error.

Fra

**Comment by jefftrull** [comment]
It is attached to your first message :)  I improvised some additional components to satisfy the references.  I can parse this DEF now, although not all the properties are stored.  The nicolati_bugs branch has the new code.  Can you try it?

**Comment by nicolati** [comment]
Hi Jeff,
ok! Now I saw it! :)
As I said, it depends on the context.
If we are forcing the fact that the only component instantiator is the DEF, then you are right, the DEF is wrong. I modified it from a real one, since I cannot share real names, but I didn't think about that. Good catch! :)
Let me have some time to try it out.

Thank you,
Fra
